### PR TITLE
delete 10 wrong items from Gordok Mastiff's loot table.

### PR DESCRIPTION
**Pull request by baranzadeoglu**
deletes 10 wrong items from Gordok Mastiff's loot table.

## Changes Proposed:
-  removes 10 items from Mastiff's loot table.
-  
image

## Issues Addressed:

- https://github.com/chromiecraft/chromiecraft/issues/2688
- https://github.com/azerothcore/azerothcore-wotlk/issues/9939
- https://github.com/chromiecraft/chromiecraft/issues/2707 

## Tests Performed:

- 
- 

## How to Test the Changes:

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Shineslippers** [comment]
I guess you have to remove 18255, 18266, 18297, 18365 from any CLT not only Gordok Mastiff

**Comment by baranzadeoglu** [comment]
> I guess you have to remove 18255, 18266, 18297, 18365 from any CLT not only Gordok Mastiff

possibly. I was just trying to fix mastiff issue when I encountered those

**Comment by Shineslippers** [comment]
If you will remove all those items from CLTs there except 18255(Runn Tum Tuber) from 14354(Pusillin) you can close https://github.com/chromiecraft/chromiecraft/issues/2707 otherwise i have to create new report on AC for it.

**Comment by acidmanifesto** [comment]
Is there a reference with these loot removals?

**Comment by Shineslippers** [comment]
Wowhead and wowwiki both are confirming it

**Comment by baranzadeoglu** [comment]
> If you will remove all those items from CLTs there except 18255(Runn Tum Tuber) from 14354(Pusillin) you can close [chromiecraft/chromiecraft#2707](https://github.com/chromiecraft/chromiecraft/issues/2707) otherwise i have to create new report on AC for it.

here you go.

**Comment by acidmanifesto** [comment]
> > If you will remove all those items from CLTs there except 18255(Runn Tum Tuber) from 14354(Pusillin) you can close [chromiecraft/chromiecraft#2707](https://github.com/chromiecraft/chromiecraft/issues/2707) otherwise i have to create new report on AC for it.
> 
> here you go.

No that isnt what I mean. What datamine db or sniff is referencing this?

**Comment by Shineslippers** [comment]
As i said above Wowhead, wowwiki and logic is enought to report it. I can't add sniffs to it.

**Comment by acidmanifesto** [comment]
Looking over compared sniffs and udb datamines I see the following items are not suppose to be present in the creature Gordok Mastiff:
5263 Pocket Lint - Is only dropped by Mizzle the Crafty NPC, not other NPCS.
18255 Runn Tum Tuber - Is only dropped by Pusillin, not other NPCs.
18266 Gordok Courtyard Key - This is only obtained thru Fengus's Chest chest and is for some reason being slapped in a few npc loot tables. Alot of Pservers dump this on npcs due to low pop as the chest is only found on Alterac Valley and Dire Maul. 
18297 Thornling Seed - This is only obtained in warpwood pods not thru npc drop, never was obtainable thru npc drops in any expansion.

File: data/sql/updates/pending_db_world/rev_1640946362151198685.sql [hunk 1]
```diff
@@ -0,0 +1,6 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1640946362151198685');
+
+DELETE FROM `creature_loot_template` WHERE (`Entry` = 13036) AND (`Item` BETWEEN 18255 AND 18533);
```
Comment (acidmanifesto) [inline data/sql/updates/pending_db_world/rev_1640946362151198685.sql:3]: Please adjust per my information based on sniffs and udb comparisons from the sniff team collective.

https://github.com/azerothcore/azerothcore-wotlk/pull/9947#issuecomment-1003409087

**Review by acidmanifesto — CHANGES_REQUESTED** [review]

File: data/sql/updates/pending_db_world/rev_1640946362151198685.sql [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+DELETE FROM `creature_loot_template` WHERE (`Item` IN (18255, 18266, 18297, 18365)) AND (`Entry` != 14354);
```
Comment (acidmanifesto) [inline data/sql/updates/pending_db_world/rev_1640946362151198685.sql:4]: If you need us too, we can verify these items before we do a blanket all delete that isnt entry 14354 with those four items.

**Review by acidmanifesto — CHANGES_REQUESTED** [review]

**Comment by acidmanifesto** [comment]
Table 'creature_loot_template' Entry 7549 does not exist but it is used by Creature 7549
image

**Comment by acidmanifesto** [comment]
You can disregard that one ss i sent. I did a pr that fixed that independently. But please review your sql when you have the time.

**Comment by baranzadeoglu** [comment]
> Please adjust it based on the data provided. All others for Gordok Mastiff is correct in the azerothcore database.

What do you mean?  There is no way loot like Ogre Forged Hauberk is correct. They are supposed to drop from the chest.

**Comment by acidmanifesto** [comment]
> > Please adjust it based on the data provided. All others for Gordok Mastiff is correct in the azerothcore database.
> 
> What do you mean? There is no way loot like Ogre Forged Hauberk is correct. They are supposed to drop from the chest.

Can you show me where you source is that states that? So far our reference source with the sniff team is the udb made of verified consolidated sniffs of a per loot per kill of 1k to 10k kill consolidated. In this case since the ratio is in the hundredth place of being 0.02 the kill cumlative on the retail verified sniff was 1k total via all collected and combined sniffs in a data mining faction at the time of retail for the 335 verified 12340 retail.
The item is verified to drop from the npc and also shares a loot container reference of 12007 which attaches to Gordok Tribute Gameobject entry 179564.

**Comment by baranzadeoglu** [comment]
> > > Please adjust it based on the data provided. All others for Gordok Mastiff is correct in the azerothcore database.
> > 
> > 
> > What do you mean? There is no way loot like Ogre Forged Hauberk is correct. They are supposed to drop from the chest.
> 
> Can you show me where you source is that states that? 

https://tbc.wowhead.com/npc=13036/gordok-mastiff#drops;mode:normal;0-3+20+17+1

you can check live and tbc classic. No indication it should drop.

**Comment by acidmanifesto** [comment]
> > > > Please adjust it based on the data provided. All others for Gordok Mastiff is correct in the azerothcore database.
> > > 
> > > 
> > > What do you mean? There is no way loot like Ogre Forged Hauberk is correct. They are supposed to drop from the chest.
> > 
> > 
> > Can you show me where you source is that states that?
> 
> https://tbc.wowhead.com/npc=13036/gordok-mastiff#drops;mode:normal;0-3+20+17+1
> 
> you can check live and tbc classic. No indication it should drop.

The best way to explain this in this case to have a better understanding where I am coming from is for me to start explaining what a sniff and udb is and how it is different from wowhead, and also how it covers gameplay as you suggested.

**Sniffer** is short and slang for a A packet sniffer — also known as a packet analyzer, protocol analyzer or network analyzer — is a piece of hardware or software used to monitor network traffic. Sniffers work by examining streams of data packets that flow between computers on a network as well as between networked computers and the larger Internet. Every packet exchange between you and the server is logged into a file call a sniff or sniff file. 

**Sniffs** comrprises of all information and telementry from everything that happens on a online session and can only be covered on retail due to the unique set of the bnet login tokens, althought pservers can be sniffed, it would instantly recognized and would not be able to be parsed (interpeted) with our sniff parser software.

**UDB** is a univiersal data base. It was the original database that all emulators were based on at one time and is currently only comprised of sniff retails only thru a practice of sniff discipline where only retail sniffs of select expansions are covered. 

**Why we are not taking wowhead as a only source of loot% anymore?**
Do not get me wrong, it could be great for reference for the other information it does provided right here and there. How ever their source of drops and drop % comes from a wowhead addon.
https://www.wowhead.com/client
image

Wowhead does not have a way in verifying loot drops and loot itself from verified sniffs directly with the blizz server.
Wowhead uses a client\addon. Which collects data automatically and as such, gets diluted from repacks and pserver modified loot drops. Which is evidence by alot of loot but the most infamouse is the strider meat coming from copper ore.
image

**Loot and Azerothcore**
Since new ways of verification became open and available by Trinitycore, the sniff team collective was reopened and new guides have been writtened.
https://www.azerothcore.org/wiki/sniffing-and-parsing
https://trinitycore.info/en/install/Sniffs

We utlized the UDB since it is comprised of mass parsing thru verified retailed sniffs and not from a addon collection. These sniffs cumlative of multiple sessions from multiple areas and same areas at various times and various actions. 
We use this to consolidate loot %. **The UDB is the player thru consolidated sniffs.**

If the loot % was a whole number only then it was determined based on 100 cumlative Loots.
If it was based on a tenth decimal value it was then determined based on 1,000 cumlative loots.
If it was based on a hundredth decimal value it was then determined based on 10,000 cumlative  loots.

Obtainin loot % sniff is consider the only verified way without the dilution.

**Why does azerothcore need to have their loot fixed in various spots?**
Azerothcore was originally a p-server source that went open source.

**Why do we not import the UDB into Azerothcore**
The UDB is lacking comments for over 90% of the stats collected from mass parsing.
It does not cover conditional loots based on condition values.

**Comment by baranzadeoglu** [comment]
thank you for the long explanation which refutes the source. However I am not convinced that there is a good reason why just _some_ of the tribute chest drops would drop from a non-elite trash. It makes no game design sense. Plus there has been a lot of false data on this database (such as "bolt of ..." dropping) with the exact same 0.02% chance. I am not convinced these items should drop from this dog. 

Plus your refutation of wowhead is based on false positives i.e. the addon registering a certain drop to an incorrect object. However, in this case we have loot that does not appear. This could only be possible with a false negative. i.e. wowhead addon has to go wrong exactly at the moment when any of the 6 items with 0.02% chance drops.

**Comment by acidmanifesto** [comment]
> thank you for the long explanation which refutes the source. However I am not convinced that there is a good reason why just _some_ of the tribute chest drops would drop from a non-elite trash. It makes no game design sense. Plus there has been a lot of false data on this database (such as "bolt of ..." dropping) with the exact same 0.02% chance. I am not convinced these items should drop from this dog.

Which database are your referring to? The UDB database is based on sniffs only.

> Plus your refutation of wowhead is based on false positives i.e. the addon registering a certain drop to an incorrect object. However, in this case we have loot that does not appear. This could only be possible with a false negative. i.e. wowhead addon has to go wrong exactly at the moment when any of the 6 items with 0.02% chance drops.

Not entirely certain that is how false positives work or is definitioned in this case. Wowhead's client\addon does not verify token and login association that sniffs do in verifying retail only connections.

**Comment by acidmanifesto** [comment]
Now going back to main topic of the PR here:
1. Your pr is label to fixed the drop of one npc
2. Your PR itself deletes loot other then the NPC ID at line 4 which this pr is titled for. We need this PR to be dedicated to the npc or the loot itself, not both.
It is in regards to line four of your sql:
`DELETE FROM `creature_loot_template` WHERE (`Item` IN (18255, 18266, 18297, 18365)) AND (`Entry` != 14354);`

3. I gave you what we had wrong in our db based on a audit that was triggered from your PR.
4. Please provide sniffs if you feel like the information is wrong based on the loot %, if not please correct based on the UDB values provided to you from the sniff team based on the audit your pr triggered.

**Comment by baranzadeoglu** [comment]
> > thank you for the long explanation which refutes the source. However I am not convinced that there is a good reason why just _some_ of the tribute chest drops would drop from a non-elite trash. It makes no game design sense. Plus there has been a lot of false data on this database (such as "bolt of ..." dropping) with the exact same 0.02% chance. I am not convinced these items should drop from this dog.
> 
> Which database are your referring to? The UDB database is based on sniffs only.

The database azerothcore is using.
 
> > Plus your refutation of wowhead is based on false positives i.e. the addon registering a certain drop to an incorrect object. However, in this case we have loot that does not appear. This could only be possible with a false negative. i.e. wowhead addon has to go wrong exactly at the moment when any of the 6 items with 0.02% chance drops.
> 
> Not entirely certain that is how false positives work or is definitioned in this case. Wowhead's client\addon does not verify token and login association that sniffs do in verifying retail only connections.

You are saying wow-head can be unreliable because it registered certain items incorrectly (like your strider meat example). But if the drop rate were correct, wow-head addon needs to NOT register exactly when they drop. Highly unlikely.

Let alone this drop in itself makes no sense.

**Comment by baranzadeoglu** [comment]
> Now going back to main topic of the PR here:
> 
>     1. Your pr is label to fixed the drop of one npc
> 
>     2. Your PR itself deletes loot other then the NPC ID at line 4 which this pr is titled for. We need this PR to be dedicated to the npc or the loot itself, not both.
>        It is in regards to line four of your sql:
> 
> 
> ```
> DELETE FROM `creature_loot_template` WHERE (`Item` IN (18255, 18266, 18297, 18365)) AND (`Entry` != 14354);
> ```
> 
>     3. I gave you what we had wrong in our db based on a audit that was triggered from your PR.
> 
>     4. Please provide sniffs if you feel like the information is wrong based on the loot %, if not please correct based on the UDB values provided to you from the sniff team based on the audit your pr triggered.

1) and 2) you can look into commit history and the suggestion from @Shineslippers .
3) and 4) I will not request a change (pull) that I believe is incorrect. Feel free to close this one and write your own.

**Comment by acidmanifesto** [comment]
> > > thank you for the long explanation which refutes the source. However I am not convinced that there is a good reason why just _some_ of the tribute chest drops would drop from a non-elite trash. It makes no game design sense. Plus there has been a lot of false data on this database (such as "bolt of ..." dropping) with the exact same 0.02% chance. I am not convinced these items should drop from this dog.
> > 
> > 
> > Which database are your referring to? The UDB database is based on sniffs only.
> 
> The database azerothcore is using.

As mentioned earlier. The database was when azerothcore was a pserver db, adjustments will need to be made as audited and found.

> > > Plus your refutation of wowhead is based on false positives i.e. the addon registering a certain drop to an incorrect object. However, in this case we have loot that does not appear. This could only be possible with a false negative. i.e. wowhead addon has to go wrong exactly at the moment when any of the 6 items with 0.02% chance drops.

No i said the loot and drop are indiscrimately collected even from pserver and repack users which makes it diluted due to their custom drops and %.

> > 
> > 
> > Not entirely certain that is how false positives work or is definitioned in this case. Wowhead's client\addon does not verify token and login association that sniffs do in verifying retail only connections.
> 
> You are saying wow-head can be unreliable because it registered certain items incorrectly (like your strider meat example). But if the drop rate were correct, wow-head addon needs to NOT register exactly when they drop. Highly unlikely.

Again, No i said the loot and drop are indiscrimately collected even from pserver and repack users which makes it diluted due to their custom drops and %.

> Let alone this drop in itself makes no sense.

Gut feelings need to have supporting evidence based on verifiable sources.

**Comment by baranzadeoglu** [comment]
> Gut feelings need to have supporting evidence based on verifiable sources.
common sense =/= gut feeling. 
Again,

"I will not request a change (pull) that I believe is incorrect. Feel free to close this one and write your own."

**Comment by baranzadeoglu** [comment]
> " No i said the loot and drop are indiscrimately collected even from pserver and repack users which makes it diluted due to their custom drops and %."

This do not account for items that are supposed to drop being missing on wowhead. Diluted =/= non-existent. Pserver data or repacks can only add more drops or dilute the drop rates. They can not remove items that you are alleging drops.

**Comment by acidmanifesto** [comment]
> > Gut feelings need to have supporting evidence based on verifiable sources.
> 
> common sense =/= gut feeling. Again,
> 
> "I will not request a change (pull) that I believe is incorrect. Feel free to close this one and write your own."

Closed at users request.

**Comment by Shineslippers** [comment]
There is too hot without a decent reason. Sorry for my suggestion we have to do it different way. My bad.

Created report for other items https://github.com/azerothcore/azerothcore-wotlk/issues/9957

This Pr is still can be used for https://github.com/azerothcore/azerothcore-wotlk/issues/9939

**Comment by acidmanifesto** [comment]
> Created report for other items #9957
This will have to be a compelte zone\instance\area audit for loot and will take some time.

> This Pr is still can be used for #9939
We have a correction pr based on audit for this now in a proper formate that focused only on that npc in particular.

**Pull request closed without merging**